From 48f9bd3f8b2e75c2bc1a3d7ef6dcea91c3debeed Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 18:35:13 -0400
Subject: [ethernet] bnx2: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate

Message-id: <1409942115-26825-7-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91499
O-Subject: [RHEL7.1 PATCH 6/8] bnx2: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate
Bugzilla: 1092495
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1092495

part of upstream commit 537fae0101c8853abb52136788173dde74b9d1e9
("net: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate")

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2.c b/drivers/net/ethernet/broadcom/bnx2.c
index 0ab8370..67d2b00 100644
--- a/drivers/net/ethernet/broadcom/bnx2.c
+++ b/drivers/net/ethernet/broadcom/bnx2.c
@@ -6916,8 +6916,8 @@ bnx2_get_settings(struct net_device *dev, struct ethtool_cmd *cmd)
   }
  }
  else {
-  ethtool_cmd_speed_set(cmd, -1);
-  cmd->duplex = -1;
+  ethtool_cmd_speed_set(cmd, SPEED_UNKNOWN);
+  cmd->duplex = DUPLEX_UNKNOWN;
  }
  spin_unlock_bh(&bp->phy_lock);
 
-- 
1.7.1